From 4b5d870e2fb679ced355482afbd34d9a2ed313a1 Mon Sep 17 00:00:00 2001
From: wfeng6 <wei.feng@intel.com>
Date: Tue, 16 Oct 2012 16:32:57 +0800
Subject: Change the seek mode of video editor preview to PREVIOUS_SYNC

BZ: 61596

The original seek mode SEEK_CLOSEST, will cause the OMXCodec to
skip the decoded buffer which has earlier timestamp than the
seek time. In some situations, the during between two I frames
is long. The OMXCodec will keep skipping the decoded buffer,
and no output is given. This will lead to the overtime error
and make the video editor application report errors.

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: Yes

Change-Id: I02c1ca32132c26a0152e1812f7983c0527dc8c9b
Orig-Change-Id: Ib4a3e9d9ced6541139f1b452e8a545f36adfaa2a
Signed-off-by: wfeng6 <wei.feng@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 libvideoeditor/lvpp/PreviewPlayer.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libvideoeditor/lvpp/PreviewPlayer.cpp b/libvideoeditor/lvpp/PreviewPlayer.cpp
index 2bd9f84..4d41615 100755
--- a/libvideoeditor/lvpp/PreviewPlayer.cpp
+++ b/libvideoeditor/lvpp/PreviewPlayer.cpp
@@ -1470,7 +1470,7 @@ status_t PreviewPlayer::readFirstVideoFrame() {
                     mSeekTimeUs / 1E6);
 
             options.setSeekTo(
-                    mSeekTimeUs, MediaSource::ReadOptions::SEEK_CLOSEST);
+                    mSeekTimeUs, MediaSource::ReadOptions::SEEK_PREVIOUS_SYNC);
         }
         for (;;) {
             status_t err = mVideoSource->read(&mVideoBuffer, &options);
-- 
1.7.4.1

